src/schema/deepmerge.js

```diff
@@ -32,11 +29,16 @@ function oldArrayMerge(target, source, optionsArgument) {
   return destination;
 }
 
-export default (a, b, opts) => {
-  if (opts) {
-    throw new Error(
-      'opts are not supported, use the deepmerge package directly'
-    );
-  }
-  return merge(a, b, { arrayMerge: oldArrayMerge });
+function mergeSimpleArrays(target, source) {
+  return [...target, ...source].filter(
+    (element, index, array) => array.indexOf(element) === index
+  );
+}
+
+export const deepmerge = (a, b) => {
+  return merge(a, b, { arrayMerge: mergeSimpleArrays });
+};
+
+export const deepmergeWithComplexArrays = (a, b) => {
```

Comment:
So, I'm thinking about other possible names that we could use to make it a bit clearer what the expected behavior would be, I think that "complex" and "simple" arrays are not making it clear enough what is the outcome that we expect from these two functions.

How about using the following names?

- `deepmerge` -> `deepmerge`
  this would be the only actual deepmerge, and the array are going to be concatenated

- `deepmergeWithComplexArray` -> `patch` or `deeppatch` (or `deepPatch` or `mergePatch`)
  This will be the "patching" helpers used to apply our "addons linter patches" to the imported schemas, and this is always "patching" the first array using the items in the second array, instead of "concatenating" them
 
- `mergeSimpleArrays` -> `concatArrays` (or `concatenateArrays`)
- `mergeComplexArrays` -> `patchArrays`

@EnTeQuAk How it sounds to you?

Reply:
I completely agree, fixed in dc639e2c